### PR TITLE
Docs for DBSCAN converted

### PR DESCRIPTION
**Pull request by outoftardis**
# Description

Converted documentation for DBSCAN (see [output](https://outoftardis.github.io/daal/dev_guide/algorithms/dbscan/index.html))

### CODE REVIEW

File: docs/source/dev_guide/algorithms/dbscan/distributed-steps/step-5.rst [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+   :header-rows: 1
+
+   * - Partial Result ID
+     - Result
+   * - ``partitionedHaloData``
+     - Pointer to the collection of ``nBlocks`` numeric tables with :math:`p` columns and arbitrary number of rows containing observations
+       from current node that should be used as halo observations on each node.
+       Numeric tables in collection ordered by the identifiers of initial block of nodes.
+
+   * - partitionedHaloDataIndices
```
Comment (vepifanov) [inline docs/source/dev_guide/algorithms/dbscan/distributed-steps/step-5.rst]: This should be put in quotation marks.

Comment (outoftardis) [reply docs/source/dev_guide/algorithms/dbscan/distributed-steps/step-5.rst]: yep, thanks, I'm fixing those things at the moment

File: docs/source/dev_guide/algorithms/dbscan/distributed-steps/step-9.rst [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+For more details, :ref:`algorithms`.
+
+.. list-table::
+   :widths: 10 60
+   :header-rows: 1
+
+   * - Result ID
+     - Result
+   * - ``step9NClusters``
+     - Pointer to :math:`1 \times 1` numeric table containing the flag indicating that the clustering process is finished on all nodes.
```
Comment (vepifanov) [inline docs/source/dev_guide/algorithms/dbscan/distributed-steps/step-9.rst]: Here is actually a mistake both here and in dev. guide.
This should be:
Pointer to :math:`1 \times 1` numeric table containing the number of clusters found on all nodes.

File: docs/source/dev_guide/algorithms/dbscan/distributed-steps/step-8.rst [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+       .. include:: distributed-steps/includes/default_result_numeric_table.rst
+
+   * - step8NClusters
+     - Pointer to :math:`1 \times 1` numeric table containing the current number of clusters found on the local node.
+
+       .. include:: distributed-steps/includes/default_result_numeric_table.rst
+
+   * - step8Queries
+     - Pointer to the collection of nBlocks numeric tables with 3 columns and arbitrary number of rows
```
Comment (vepifanov) [inline docs/source/dev_guide/algorithms/dbscan/distributed-steps/step-8.rst]: nBlocks should be put in quotation marks.

File: docs/source/dev_guide/algorithms/dbscan/distributed-steps/step-10.rst [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+       .. include:: distributed-steps/includes/default_result_numeric_table.rst
+
+   * - ``step10FinishedFlag``
+     - Pointer to :math:`1 \times 1` numeric table containing the flag indicating that the clusters numeration process is finished for current node.
+
+       .. include:: distributed-steps/includes/default_result_numeric_table.rst       
+
+   * - ``step10Queries``
+     - Pointer to the collection of nBlocks numeric tables with 4 columns and arbitrary number of rows containing clusters numeration queries that should be processed on each node.
```
Comment (vepifanov) [inline docs/source/dev_guide/algorithms/dbscan/distributed-steps/step-10.rst]: nBlocks should be put in quotation marks.

File: docs/source/dev_guide/algorithms/dbscan/distributed-steps/step-11.rst [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+       .. include:: distributed-steps/includes/default_result_numeric_table.rst
+
+   * - ``step11FinishedFlag``
+     - Pointer to :math:`1 \times 1` numeric table containing the flag indicating that the clusters numeration process is finished for current node.
+
+       .. include:: distributed-steps/includes/default_result_numeric_table.rst
+
+   * - ``step11Queries``
+     - Pointer to the collection of nBlocks numeric tables with 4 columns and arbitrary number of rows containing clusters numeration queries that should be processed on each node.
```
Comment (vepifanov) [inline docs/source/dev_guide/algorithms/dbscan/distributed-steps/step-11.rst]: nBlocks should be put in quotation marks.

File: docs/source/dev_guide/algorithms/dbscan/distributed-steps/step-12.rst [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+For more details, :ref:`algorithms`.
+
+.. list-table::
+   :widths: 10 60
+   :header-rows: 1
+
+   * - Partial Result ID
+     - Result
+   * - ``assignmentQueries``
+     - Pointer to the collection of nBlocks numeric tables with 2 columns and arbitrary number of rows
```
Comment (vepifanov) [inline docs/source/dev_guide/algorithms/dbscan/distributed-steps/step-12.rst]: nBlocks should be put in quotation marks.

**Comment by outoftardis** [comment]
@vepifanov by the way, what'd you say for a Details page to have definitions in a form of glossary? it's possible then to link to the definitions there from anywhere
image

**Comment by vepifanov** [comment]
Details page looks nice!

**Comment by outoftardis** [comment]
I've updated the [output ](https://outoftardis.github.io/daal/dev_guide/algorithms/dbscan/index.html) with all the changes.

**Comment by outoftardis** [comment]
make test

**Comment by outoftardis** [comment]
@napetrov I don't think errors in CI are caused by my commits, could you check?

**Comment by napetrov** [comment]
I also don't think that they are related.  Let's restart to double check . I will merge this later today. ok?

**Comment by napetrov** [comment]
restart

**Pull request merged**